qt/python/mantidqt/widgets/superplot/presenter.py

```diff
@@ -0,0 +1,506 @@
+        self._model.add_workspace(added_workspace)
+        self._update_list()
+        self._view.set_selection(selection)
+        self._update_plot()
+
+    def on_del_button_clicked(self, ws_name=None):
+        """
+        Triggered when the del button is pressed. This function removes the
+        selected workspace from the selection list.
+        """
+        selection = self._view.get_selection()
+        if ws_name is None:
+            selected_workspaces = selection.copy()
+        else:
+            selected_workspaces = [ws_name]
+        for selected_workspace in selected_workspaces:
+            self._model.del_workspace(selected_workspace)
+        self._update_list()
+        if not self._model.is_bin_mode() and not self._model.is_spectrum_mode():
+            mode = self._view.get_mode()
+            self._view.set_available_modes([self.SPECTRUM_MODE_TEXT,
+                                            self.BIN_MODE_TEXT])
+            self._view.set_mode(mode)
+        self._view.set_selection(selection)
+        self._update_spectrum_slider()
+        self._update_plot()
+
+    def _update_spectrum_slider(self):
+        """
+        Update the spectrum slider and spinbox to match the selected workspaces.
+        """
+        selection = self._view.get_selection()
+        if not selection:
+            self._view.set_spectrum_slider_position(0)
+            self._view.set_spectrum_slider_max(0)
+            self._view.set_spectrum_spin_box_value(0)
+            self._view.set_spectrum_spin_box_max(0)
+            self._view.set_spectrum_disabled(True)
+            return
+        maximum = None
+        position = None
+        mode = self._view.get_mode()
+        for ws_name in selection:
+            ws = mtd[ws_name]
+            for sp in selection[ws_name]:
+                if position is None:
+                    position = sp
+                elif position != sp:
+                    self._view.set_spectrum_slider_position(0)
+                    self._view.set_spectrum_slider_max(0)
+                    self._view.set_spectrum_spin_box_value(0)
+                    self._view.set_spectrum_spin_box_max(0)
+                    self._view.set_spectrum_disabled(True)
+                    return
+            if mode == self.SPECTRUM_MODE_TEXT:
+                value = ws.getNumberHistograms()
+            else:
+                value = ws.blocksize()
+            if maximum is None:
+                maximum = value
+            elif value < maximum:
+                maximum = value
+        if position is None or position >= maximum:
+            position = 0
+        self._view.set_spectrum_disabled(False)
+        self._view.set_spectrum_slider_max(maximum - 1)
+        self._view.set_spectrum_slider_position(position)
+        self._view.set_spectrum_spin_box_max(maximum - 1)
+        self._view.set_spectrum_spin_box_value(position)
+
+    def _update_hold_button(self):
```

Comment:
This function also needs to be called also immediately after the button is pressed (and the spectrum is held) so that its state is correct (maybe in `_on_hold()`).

Additionally, I think it would be a lot more intuitive to change the behaviour of the button so that the text chages from "add" to "remove", rather than the button being checked or unchecked. It is not immediately obvious that the button's check state changes if a spectrum is "held".

Reply:
This works now